internal/tracegen/worker_test.go

```diff
@@ -16,30 +16,53 @@ import (
 )
 
 func Test_SimulateTraces(t *testing.T) {
-	logger, buf := testutils.NewLogger()
-	tp := sdktrace.NewTracerProvider()
-	tracers := []trace.Tracer{tp.Tracer("stdout")}
-	wg := sync.WaitGroup{}
-	wg.Add(1)
-	var running uint32 = 1
-
-	worker := &worker{
-		logger:  logger,
-		tracers: tracers,
-		wg:      &wg,
-		id:      7,
-		running: &running,
-		Config: Config{
-			Traces:   7,
-			Duration: time.Second,
-			Pause:    time.Second,
-			Service:  "stdout",
-			Debug:    true,
-			Firehose: true,
+	tests := []struct {
+		name           string
+		pause          time.Duration
+		childSpans     int
+		expectedOutput string
+	}{
+		{
+			name:           "no pause",
+			pause:          0,
+			childSpans:     1,
+			expectedOutput: `{"level":"info","msg":"Worker 7 generated 7 traces"}` + "\n",
```

Comment:
if these are the same for all tests, no need to include them as params, it only increases cognitive overhead for understanding the test